### PR TITLE
Remove comm:dead and comm:live events

### PR DESCRIPTION
**Pull request by SylvainCorlay**
This is an attempt to remove the `comm:[dead/live]` event bubbling mechanism.

### CODE REVIEW

File: widgetsnbextension/src/extension.js [hunk 1]
```diff
@@ -43,6 +43,13 @@ var handle_cell = function(cell) {
     if (cell.cell_type==='code') {
         var area = new widgetarea.WidgetArea(cell);
         cell.widgetarea = area;
+        var died = function () {
+            area._widget_dead();
+        }
+        cell.notebook.events.on('kernel_disconnected.Kernel', died);
+        cell.notebook.events.on('kernel_killed.Kernel', died);
+        cell.notebook.events.on('kernel_restarting.Kernel', died);
+        cell.notebook.events.on('kernel_dead.Kernel', died);
```
Comment (SylvainCorlay) [inline widgetsnbextension/src/extension.js]: I should probably call `off` on disposal of the widget area.

Comment (jdfreder) [inline widgetsnbextension/src/extension.js]: +1, I don't think we were doing that before, but it's probably good to add if it's not too much effort.

Comment (jdfreder) [inline widgetsnbextension/src/extension.js]: Oh, I see, we were doing it before

Comment (jdfreder) [inline widgetsnbextension/src/extension.js]: Can you add it in a cb for the widget dispose event?

Comment (SylvainCorlay) [inline widgetsnbextension/src/extension.js]: You mean the disposal of the area?

File: widgetsnbextension/src/widgetarea.js [hunk 2]
```diff
@@ -96,10 +92,6 @@ WidgetArea.prototype._create_elements = function() {
                 for (var i = 0; i < that.widget_views.length; i++) {
                     var view = that.widget_views[i];
                     view.remove();
-
-                    // Remove widget live events.
-                    view.off('comm:live', that._widget_live);
-                    view.off('comm:dead', that._widget_dead);
```
Comment (jdfreder) [inline widgetsnbextension/src/widgetarea.js]: @SylvainCorlay I read quickly and mistaked these lines as the old unregister point.

File: widgetsnbextension/src/extension.js (hunk 1 above)
Comment (jdfreder) [inline widgetsnbextension/src/extension.js]: Yes, sorry

**Comment by SylvainCorlay** [comment]
This is ready for review.

**Comment by jdfreder** [comment]
Thanks!

**Pull request merged**